lib/ansible/module_utils/basic.py

```diff
@@ -1832,9 +1834,19 @@ def safe_eval(self, value, locals=None, include_exceptions=False):
     def _check_type_str(self, value):
         if isinstance(value, string_types):
             return value
-        # Note: This could throw a unicode error if value's __str__() method
-        # returns non-ascii.  Have to port utils.to_bytes() if that happens
-        return str(value)
+
+        # Ignore, warn, or error when converting to a string.
+        # The current default is to warn. Change this in Anisble 2.12 to error.
+        common_msg = 'The value {0!r} (type {0.__class__.__name__}) in a string field was converted to {1!r} (type string). {2}'
```

Comment:
That would be helpful, but it would required changing the method signature. I figured that displaying the value is just about as useful as the parameter name since you should be able to look at the task and find the value just as easily as you can find the parameter name.